### PR TITLE
fix: visibility for impl methods

### PR DESCRIPTION
**Pull request by asterite**
# Description

## Problem

Part of #4515

Apparently in #6179 I only made it work for calls like `foo::Bar::baz()` but no checks were done for method calls (where `self` is involved).

## Summary

This PR now warns when calling private or `pub(crate)` methods, either on structs or primitive types, when they are not accessible from the current module.

We also now don't suggest non-accessible methods from LSP.

## Additional Context

I also removed an unused global that was introduced some time ago.

## Documentation

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: compiler/noirc_frontend/src/hir_def/types.rs [hunk 1]
```diff
@@ -983,6 +983,34 @@ impl Type {
...
+            | Type::Function(..)
+            | Type::Tuple(..) => true,
+            Type::Alias(alias_type, generics) => {
+                alias_type.borrow().get_type(&generics).is_primitive()
+            }
+            Type::MutableReference(typ) => typ.is_primitive(),
+            Type::Struct(..)
+            | Type::TypeVariable(..)
+            | Type::TraitAsType(..)
+            | Type::NamedGeneric(..)
```
Comment (jfecher) [inline compiler/noirc_frontend/src/hir_def/types.rs:1005]: type variables & named generics can be primitive if they're bound to one

**Review by jfecher — COMMENTED** [review]
One thing, otherwise LGTM

File: compiler/noirc_frontend/src/hir_def/types.rs (hunk 1 above)
Comment (asterite) [reply compiler/noirc_frontend/src/hir_def/types.rs:1005]: I see. I was about to fix this but I noticed there's a `follow_bindings()` call at the beginning. Would that take care of this?

Comment (jfecher) [reply compiler/noirc_frontend/src/hir_def/types.rs:1005]: Ah, I missed that as well. It's probably fine then

**Review by jfecher — APPROVED** [review]

**Pull request merged**